src/variable.cpp

```diff
@@ -118,4 +131,39 @@ std::string Variable::getInitialValue() const
     return mPimpl->mInitialValue;
 }
 
+void Variable::setInterfaceType(Variable::INTERFACE_TYPES interfaceType)
+{
+    mPimpl->mInterfaceType = interfaceType;
+}
+
+Variable::INTERFACE_TYPES Variable::getInterfaceType() const
+{
+    return mPimpl->mInterfaceType;
+}
+
+std::string interfaceTypeToString(Variable::INTERFACE_TYPES interfaceType)
+{
+    std::string str = "";
+    switch (interfaceType) {
+    case Variable::INTERFACE_TYPE_PRIVATE: {
+        str = "private";
+        break;
+    }
+    case Variable::INTERFACE_TYPE_PUBLIC: {
+        str = "public";
+        break;
+    }
+    case Variable::INTERFACE_TYPE_PUBLIC_AND_PRIVATE: {
+        str = "public_and_private";
+        break;
+    }
+    case Variable::INTERFACE_TYPE_NONE: {
```

Comment:
From your reordering in ```prefixToString()```, I thought you wanted to have everything alphabetically ordered? If so, ```Variable::INTERFACE_TYPE_NONE``` ought to be first.